Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-327.28.3.el7.x86_64)
 GCC:gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
Sha: 445ce66b9f55739231848d6a5501bc3eb48494a9
Build: success
Build: success
Install HPCC Platform
Number of Error(s): 4
src/components/Hello.tsx(8,17): error TS2559: Type { children: string[]; } has no properties in common with type HTMLAttributes.

Error in add build result to gist repo :[Errno 2] No such file or directory: gists
<type exceptions.OSError> (line: 1433)

src/components/Hello.tsx(13,12): error TS2605: JSX element type Hello is not a constructor function for JSX elements.
CMake Error at esp/src/cmake_install.cmake:84 (message):
make: *** [Makefile:74: install] Error 1
CMake Error at esp/src/cmake_install.cmake:84 (message):
  npm run compile failed see above for details.
Call Stack (most recent call first):
  esp/cmake_install.cmake:46 (include)
  cmake_install.cmake:55 (include)
make: *** [Makefile:74: install] Error 1
Number of Error(s): 4
src/components/Hello.tsx(8,17): error TS2559: Type { children: string[]; } has no properties in common with type HTMLAttributes.
src/components/Hello.tsx(13,12): error TS2605: JSX element type Hello is not a constructor function for JSX elements.
CMake Error at esp/src/cmake_install.cmake:84 (message):
make: *** [Makefile:74: install] Error 1
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-08-12_21-59-04.log
CMakeFiles/CMakeError.log copied into ../CMakeError-2019-08-12_21-59-04.log
Get ECLWatch build result
ECL Watch:  Rebuilding Site
---- Build Target: /mnt/disk1/home/vamosax/smoketest/PR-12870/build/esp/src/build
---- Creating out of source build environment
---- Installing third party libraries
---- Copy resources
---- Build website
src/components/Hello.tsx(8,17): error TS2559: Type { children: string[]; } has no properties in common with type HTMLAttributes.
src/components/Hello.tsx(13,12): error TS2605: JSX element type Hello is not a constructor function for JSX elements.
  Type Hello is missing the following properties from type ElementClass: base, linkState
CMake Error at esp/src/cmake_install.cmake:84 (message):
  npm run compile failed see above for details.
Call Stack (most recent call first):
  esp/cmake_install.cmake:46 (include)
  cmake_install.cmake:55 (include)
make: *** [Makefile:74: install] Error 1
Number of Error(s): 4
src/components/Hello.tsx(8,17): error TS2559: Type { children: string[]; } has no properties in common with type HTMLAttributes.
src/components/Hello.tsx(13,12): error TS2605: JSX element type Hello is not a constructor function for JSX elements.
CMake Error at esp/src/cmake_install.cmake:84 (message):
make: *** [Makefile:74: install] Error 1
CMakeFiles/CMakeOutput.log copied into ../CMakeOutput-2019-08-12_21-59-04.log
CMakeFiles/CMakeError.log copied into ../CMakeError-2019-08-12_21-59-04.log

--- a/esp/src/eclwatch/dojoConfig.js
+++ b/esp/src/eclwatch/dojoConfig.js
@@ -56,7 +56,9 @@ function getConfig(env) {
             "codemirror": baseUrl + "/node_modules/codemirror",
             "crossfilter": baseUrl + "/node_modules/crossfilter2/crossfilter.min",
             "font-awesome": baseUrl + "/node_modules/@hpcc-js/common/font-awesome",
-            "tslib": baseUrl + "/node_modules/tslib/tslib"
+            "tslib": baseUrl + "/node_modules/tslib/tslib",
+            "react": baseUrl + "/node_modules/react/umd/react.production.min",
+            "react-dom": baseUrl + "/node_modules/react-dom/umd/react-dom.production.min"
         },
         packages: [
             {

--- a/esp/src/package.json
+++ b/esp/src/package.json
@@ -26,7 +26,8 @@
     "dev-start": "ws",
     "dev-link": "rimraf ./node_modules/@hpcc-js && mklink /J .\\node_modules\\@hpcc-js ..\\..\\..\\hpcc-js\\packages",
     "dev-unlink": "rimraf ./node_modules/@hpcc-js && npm install",
-    "test": "run-s lint"
+    "test": "run-s lint",
+    "postinstall": "rimraf ./node_modules/**/preact/dist/preact.d.ts"
   },
   "main": "src/stub.js",
   "dependencies": {
@@ -43,16 +44,20 @@
     "d3-array": "1.2.4",
     "d3-scale": "1.0.7",
     "d3-selection": "1.4.0",
+    "dijit": "1.15.0",
     "dojo": "1.15.0",
     "dojo-themes": "1.14.1",
     "dojo-util": "1.15.0",
-    "dijit": "1.15.0",
     "dojox": "1.15.0",
     "es6-promise": "4.2.6",
-    "font-awesome": "4.7.0"
+    "font-awesome": "4.7.0",
+    "react": "^16.9.0",
+    "react-dom": "^16.9.0"
   },
   "devDependencies": {
     "@types/dojo": "1.9.41",
+    "@types/react": "^16.9.1",
+    "@types/react-dom": "^16.8.5",
     "copy-webpack-plugin": "4.5.4",
     "cpx": "1.5.0",
     "css-loader": "2.1.1",

--- a/esp/src/src/Components/Example.tsx
+++ b/esp/src/src/Components/Example.tsx
@@ -1,0 +1,24 @@
+import * as React from "react";
+import { render } from "react-dom";
+
+ interface IProps {
+  message: string;
+}
+
+ export class Hello extends React.Component<IProps, {}> {
+  render() {
+    return <span>Hello from ECLWatch {this.props.message}!</span>;
+  }
+}
+
+
+ export function renderHello(target: HTMLElement, props: IProps) {
+    render(<Hello message={props.message} />, target);
+}
+// funtcional component
+// const Hello2 = (props: IProps) => {
+//   return <span>Hello from {props.message}!</span>;
+// };
+
+//add to widgets as such
+//ExampleComponent.renderHello(document.getElementById(this.id + "React"), { message: "Some Message" });

--- a/esp/src/tsconfig.json
+++ b/esp/src/tsconfig.json
@@ -14,6 +14,7 @@
         "allowJs": true,
         "skipLibCheck": true,
         "noImplicitUseStrict": true,
+        "jsx": "react",
         "lib": [
             "dom",
             "es5",

